aws/resource_aws_ecs_service.go

```diff
@@ -66,6 +68,13 @@ func resourceAwsEcsService() *schema.Resource {
 				Default:  "EC2",
 			},
 
+			"scheduling_strategy": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "REPLICA",
+			},
```

Comment:
This isn't done consistently but I find https://github.com/terraform-providers/terraform-provider-aws/blob/41a726c851769570036f30cd83ffc2b68a47c8dd/aws/resource_aws_lb_listener.go#L50-L52 is a nice convenience for users.

Not sure if it's worth validating this so we can generate failures at plan time because it comes with the trade off that if AWS adds another scheduling strategy we need a code change and release to support it.

Reply:
Will add the the `StateFunc`. Thanks for the advice @tomelliff .

Regarding the validation, that's also why I'm a bit hesitant about adding. I think it might come down to consistency, i.e. whether this provide performs validations like this somewhere else.